nav2_costmap_2d/src/costmap_2d.cpp

```diff
@@ -102,6 +102,8 @@ void Costmap2D::initMaps(unsigned int size_x, unsigned int size_y)
 {
   std::unique_lock<mutex_t> lock(*access_);
   delete[] costmap_;
+  size_x_ = size_x;
```

Comment:
You then can remove the setting of `size_x_` in the places where this method is called :smile: